src/context/WalletProvider/WalletConnectV2/Web3ModalProvider.tsx

```diff
@@ -0,0 +1,52 @@
+import { createConfig } from '@wagmi/core'
+import { EthereumClient, w3mConnectors } from '@web3modal/ethereum'
+import { Web3Modal } from '@web3modal/react'
+import { getConfig } from 'config'
+import React, { createContext, useContext } from 'react'
+import { mainnet } from 'viem/chains'
+import { viemClient } from 'lib/viem-client'
+
+type Web3ModalProviderProps = {
+  children: React.ReactNode
+}
+
+type Web3ModalContextType = {
+  ethereumClient: EthereumClient
+}
+
+export const Web3ModalContext = createContext<Web3ModalContextType | undefined>(undefined)
+
+// Initialization functions
+const createEthereumClient = () => {
+  const chains = [mainnet]
```

Comment:
Do we want to match `optionalChains` here?

Reply:
We do! Addressed in https://github.com/shapeshift/web/pull/5315/commits/bae3cf8f51be00329e9ead4e297b3eca523be7b3.